### PR TITLE
Allow auto assign population to specialists.

### PR DESCRIPTION
**Pull request by ninjatao**
https://github.com/yairm210/UnCiv/issues/255

Checked in game that AI cities now has specialists. But please help review the code. In fact this is my first attempt on Kotlin >o<

### CODE REVIEW

**Comment by yairm210** [comment]
Did you see my comments?

**Comment by ninjatao** [comment]
What comments? I see nothing in this pull request, nor in the issue.

**Comment by yairm210** [comment]
In the file changes of the pull request

**Comment by ninjatao** [comment]
screen shot 2018-12-09 at 1 35 36 pm

In here? I found nothing... Or do you mean comments in the code files?

File: core/src/com/unciv/logic/city/PopulationManager.kt [hunk 1]
```diff
@@ -87,16 +110,29 @@ class PopulationManager {
...
+            val worstJob: Stat? = specialists.toHashMap()
+                    .filter { it.value > 0 }
+                    .minBy { Automation().rankSpecialist(cityInfo.cityStats.getStatsOfSpecialist(it.key, policies), cityInfo.civInfo) }
+                    ?.key
+            var valueWorstSpecialist = 0f
+            if (worstJob != null)
+                valueWorstSpecialist = Automation().rankSpecialist(cityInfo.cityStats.getStatsOfSpecialist(worstJob, policies), cityInfo.civInfo)
+
+            //un-assign population
+            if ((valueWorstTile < valueWorstSpecialist && worstWorkedTile != null)
```
[the remark this answers is not in the document]
Comment (ninjatao) [reply core/src/com/unciv/logic/city/PopulationManager.kt]: screen shot 2018-12-09 at 6 51 40 pm

no specialists has been checked.

[the remark this answers is not in the document]
Comment (ninjatao) [reply core/src/com/unciv/logic/city/PopulationManager.kt]: Will replace it with "worstJob == null"

**Comment by ninjatao** [comment]
Updated.

**Comment by yairm210** [comment]
Solves #255

**Pull request merged**